docs/tutorials/lognamespacing.md

```diff
@@ -0,0 +1,431 @@
+be returned by calls to `log_schema()` eg. `log_schema().source_type_key()` 
+or `log_schema().timestamp_key()`.
+- Some sources allow the user to specify the field name that a given
+value will be placed in. For example, the `kafka` source will allow the
+user to specify the `topic_key` - the field name that will contain the 
+kafka `topic` the event was consumed from.  
+- Other sources just hardcode this value. For example the `dnstap` source
+creates an event with an object, most of these field names are hardcoded.
+
+#### metadata_key
+
+The name of the field when it is inserted into the Vector namespace. This 
+will be `path!("ingest_timestamp")` or `path!("source_type")`. The field names
+can be hardcoded since they are going into the Vector namespace, so conflicts
+with other field names cannot occur.
+
+It should be noted that the values for these field names are typically 
+hardcoded. With the `kafka` source, for example, it was possible to configure
+the field name that the `topic` was inserted into. In the Vector namespace
+this field name is just hardcoded to `topic`. Allowing the user to configure
+the fieldname was only necessary to prevent name conflicts with other values
+from the event. This is no longer an issue as these values are now placed in a 
+separate namespace to the event data.
+
+#### value
+
+The actual value to be placed into the field. 
+
+For the ingest timestamp this will be `chrono::Utc::now()`. Source type will be
+the `NAME` property of the `Config` struct. `NAME` is provided by the 
+`configurable_component` macro. You may need to include `use vector_config::NamedComponent;`.
+
+For batches of events, each event in the batch should use a precalculated
+`Utc::now()` so they all share the same timestamp.
+
+
+### insert_standard_vector_source_metadata(...)
+
+A utility function has been provided that can be used in a lot of cases to 
+insert both these fields into the Vector namespace:
+
+```rust
+
+  log_namespace.insert_standard_vector_source_metadata(
+      log,
+      KafkaSourceConfig::NAME,
+      Utc::now(),
+  );
+```
+
+### Source Metadata
+
+Other fields that describe the event - but are not the actual data for the event
+should go into the source metadata. Examples of source metadata are 
+
+- The Kafka topic when pulling from a Kafka stream.
+- Severity and Facility fields from a Syslog message.
+- The file path when pulling data from a file.
+
+To insert source metadata:
+
+```rust
+  log_namespace.insert_source_metadata(
+      SyslogConfig::NAME,
+      log,
+      Some(LegacyKey::Overwrite("source_id")),
+      path!("source_id"),
+      default_host.clone(),
+  );
+```
+
+Let's look at the parameters:
+
+```rust
+    pub fn insert_source_metadata<'a>(
+        &self,
+        source_name: &'a str,
+        log: &mut LogEvent,
+        legacy_key: Option<LegacyKey<impl ValuePath<'a>>>,
+        metadata_key: impl ValuePath<'a>,
+        value: impl Into<Value>,
+    ) 
+```
+
+#### source_name
```

Comment:
I think the main problem with making it a list is there are some parameters, eg.  `meaning` described later on that contains other lists, so it would get a bit confused.